spng.c

```diff
@@ -1782,9 +1782,14 @@ int spng_decode_image(spng_ctx *ctx, unsigned char *out, size_t out_size, int fm
 
     struct decode_flags f = {0};
 
+    /* We cannot properly apply all flags on "raw" data for all data types
+    additionally the format implys no additional transforms would be applied */
+    if (fmt == SPNG_FMT_RAW && flags != 0) return SPNG_EFLAGS;
+
     ret = spng_decoded_image_size(ctx, fmt, &out_size_required);
     if(ret) return ret;
     if(out_size < out_size_required) return SPNG_EBUFSIZ;
+    memset(out, 0, out_size_required);
```

Comment:
Please limit this to `SPNG_FMT_RAW`, interlaced, <8-bit images to minimize the performance impact.